code/common/os.q

```diff
@@ -7,10 +7,12 @@ ext:{`$(x?".")_(x:string x;x)[i:10h=type x]}
 del:{system("rm ";"del ")[NT],pth x}
 deldir:{system("rm -r ";"rd /s /q ")[NT],pth x}
 hdeldir:{[dirpath]
+ dirpath:@[{[x] $[10h~a:type x;x;-11h~a;string x;'`err]};dirpath;{.lg.e[`hdeldir;"Invalid data type for directory path"]}];
```

Comment:
This would make it exit the function on error without the need for the check below (you can delete line 11).
```suggestion
 dirpath:@[{[x] $[10h~a:type x;x;-11h~a;string x;'`err]};dirpath;{.lg.e[`hdeldir;"Invalid data type for directory path"];:()}];
```

But, do we want to exit like this or do we actually want to just throw the error and catch it where this function is called?

```suggestion
 dirpath:$[10h~a:type dirpath;x;-11h~a;string x;[.lg.e[`hdeldir;"Invalid data type for directory path: type ",string a];'"Invalid type"]];
```

Or if you don't want to throw you can still just exit here:

```suggestion
 dirpath:$[10h~a:type dirpath;x;-11h~a;string x;[.lg.e[`hdeldir;"Invalid data type for directory path: type ",string a];:()]];
```

Reply:
I agree that error should be thrown, no need to log error here IMO - keep it simple

```suggestion
 dirpath:$[10h~a:type dirpath;dirpath;-11h~a;string dirpath;'`type];
```

kdb devs are familiar enough with `'type` errors IMO that it's fine to throw that as the error